pkg/move/current.go

```diff
@@ -0,0 +1,72 @@
+package move
+
+import (
+	iofs "io/fs"
+	"path/filepath"
+	"regexp"
+
+	"github.com/Dynatrace/dynatrace-bootstrapper/pkg/utils/fs/symlink"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+const (
+	// example match: 1.239.14.20220325-164521
+	versionRegexp = `^(\d+)\.(\d+)\.(\d+)\.(\d+)-(\d+)$`
+	binDir        = "/agent/bin"
+	currentDir    = "current"
+)
+
+// CreateCurrentSymlink finds a the version of the CodeModule in the `targetDir` and creates a "current" symlink next to it.
+// this is needed for the nginx use-case.
+func CreateCurrentSymlink(log logr.Logger, fs afero.Fs, targetDir string) error {
+	targetBinDir := filepath.Join(targetDir, binDir)
+
+	relativeSymlinkPath, err := findVersionFromFS(log, fs, targetBinDir)
+	if err != nil {
+		log.Info("failed to get the version from the filesystem", "targetDir", targetDir)
+
+		return err
+	}
+
+	return symlink.Create(log, fs, relativeSymlinkPath, filepath.Join(targetBinDir, currentDir))
+}
+
+func findVersionFromFS(log logr.Logger, fs afero.Fs, targetBinDir string) (string, error) {
```

Comment:
Instead of walking the fs, did you consider reading `./agent/installer.version`?

Reply:
done (used what has been there since forever)

Reply:
We should consider changing that in the codemodule injection installer in the operator